core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -211,7 +211,8 @@ public boolean accept(CarbonFile file) {
         return file.getName().startsWith(SCHEMA_FILE);
       }
     });
-    if (schemaFile != null && schemaFile.length > 0) {
+    if (schemaFile != null && schemaFile.length > 0 &&
+        FileFactory.getFileType(tablePath) != FileFactory.FileType.ALLUXIO) {
```

Comment:
It is a bit strange to handle differently for Alluxio in this class, can we wrap it in different CarbonFile implementation?

Reply:
will move to another pr